src/coreclr/jit/lsrabuild.cpp

```diff
@@ -1456,6 +1456,34 @@ Interval* LinearScan::getUpperVectorInterval(unsigned varIndex)
     unreached();
 }
 
+//------------------------------------------------------------------------
+// shouldBuildUpperVectorSaveRestore - Check if vector save/restore is needed.
+//
+// Arguments:
+//    tree       - The current node being handled
+//
+// Notes: Vector save/restore is usually not needed before the call that never returns.
+//        It is also not needed if the block ends with a throw.
+bool LinearScan::shouldBuildUpperVectorSaveRestore(GenTree* tree)
+{
+    if ((tree != nullptr) && tree->IsCall())
+    {
+        if (tree->AsCall()->IsNoReturn())
+        {
+            // No point in having vector save/restore if the call will not return.
+            return false;
+        }
+    }
+
+    if ((compiler->compCurBB != nullptr) && (compiler->compCurBB->KindIs(BBJ_THROW)))
```

Comment:
Perhaps, block should a parameter, or method renamed to something stating it checks current block?

Reply:
I thought about passing block as a parameter but there are lot of call chain where I would have to pass `block`. We use `compCurBB` in earlier phases as well as `curBBNum` in LSRA to avoid doing that. As far method name, I think I have stated in the comments that it inspects the block.